### DISCUSSION TITLE
Generating sign header token

### DISCUSSION THREAD

**Discussion by Cromzinc** [DIGITALCRIMINALS/OnlyFans#464] [Q&A]
To find the sign header token you previously mentioned the following: 

```
def create_sign(session, link, sess, user_agent, text="onlyfans"):
    time = str(int(round(time2.time() * 1000-300000)))
    path = urlparse(link).path
    query = urlparse(link).query
    path = path+"?"+query
    a = [sess, time, path, user_agent, text]
    msg = "\n".join(a)
    message = msg.encode("utf-8")
    hash_object = hashlib.sha1(message)
    sha_1 = hash_object.hexdigest()
    session.headers["sign"] = sha_1
    session.headers["time"] = time
    return session
```

This does not seem to provide the correct sign token, have they changed the way the generate this value?

sign header value from the browser is formatted like this: 3403:9a6c9c8c73002eaec029b4047a20a218698b6911:a0d:62824e43
and the one I'm generating is formatted like this: 725097acd09253dce52f37af47dab2251c43328c

Thanks!

**Comment by DIGITALCRIMINAL** [comment]
Are you using Python (this function) to generate the token?

The token you generate is based on the link, sess and user agent. 
So wherever you got the token from, you'll have to use the request link and pass it to the function.

**Reply by Cromzinc** [reply]
Thanks for the quick response . 

Yeah, I was using that function. Noticed my requests were failing. So I tried to generate the same sign token the website was creating when making the call. Used fiddler to catch the call, used the same link, sess, user agent from the captured call, but get a much different sign token. 

From the request I captured, OF sign token is formatted much different than sha1.
